equinox/serialisation.py

```diff
@@ -240,8 +246,8 @@ def _deserialise(spec, x):
             def __deserialise(y):
                 return spec(f, y)
 
-            return jtu.tree_map(__deserialise, x, is_leaf=is_leaf)
+            return _ordered_tree_map(__deserialise, x, is_leaf=is_leaf)
 
-        out = jtu.tree_map(_deserialise, filter_spec, like)
-    jtu.tree_map(_assert_same, out, like, is_leaf=is_leaf)
+        out = _ordered_tree_map(_deserialise, filter_spec, like)
+    _ordered_tree_map(_assert_same, out, like, is_leaf=is_leaf)
```

Comment:
We don't need the `_assert_same` to occur in a guaranteed order. Not that it really matters, but it's not a necessary change.

Reply:
But then while deserialising it would become an issue if leaves are saved in arbitrary order but accessed in DFS?

Reply:
Sorry ignore this.

Reply:
Indeed, the `out = _ordered...` line is correct. It's the line afterwards (`_ordered_tree_map(_assert_same, ...`) that doesn't need to be ordered.

Like I say this doesn't really matter, but IMO it's a bit neater not to make an unnecessary change.